frontend/src/pages/projects/screens/detail/storage/ManageStorageModal.tsx

```diff
@@ -84,7 +84,7 @@ const ManageStorageModal: React.FC<AddStorageModalProps> = ({ existingData, isOp
       }
       return;
     }
-    const createdPvc = await createPvc(createData, namespace, { dryRun });
+    const createdPvc = await createPvc(createData, namespace, undefined, { dryRun });
```

Comment:
where does this undefined stem from?

Reply:
@shalberd If you have multiple optional params, as is the case for the createPvc function, TypeScript isn't able to compile this down to JavaScript in a meaningful way - as such the result would be the 4th param `{ dryRun }` would get passed as the third param and you'd get a TypeError at runtime - as such TypeScript won't try to infer from the types which optional param you're passing.

Passing `undefined` as the third param has the exact same practical effect as not passing an optional param.

Reply:
@alexcreasy about the existing code usepreferredStorageClass that determines storageClass name ... maybe we could move that one level up from screens/spawner/storage/usePreferredStorageClass to
screens/storage/usePreferredStorageClass, would need to modify the import line in SpawnerFooter then, but that would make it clearer that the storageClas logic is not just for SpawnerFooter / New Workbench, but also useful for the ManageStorageModal / Add New Storage. What do you think?

> Passing undefined as the third param has the exact same practical effect as not passing an optional param.

I see, yes. Well, in any case, for the Add New Storage Modal to work on clusters with no default storage Class set. we should get the StorageClassKind from logic at https://github.com/opendatahub-io/odh-dashboard/pull/2503/files#diff-919e2df35bd1299179bab541aa9609ac0dbf6bde2e5dcc0f70849180ecebec63R5 and then pass on metadata.name to here instead of undefined.

Reply:
@shalberd so this was causing an issue where the storage class isn't set when using the "Add cluster storage" feature on the project details screen. I've pushed an update that should now set this, both when using the "Create workbench modal" and the "Add cluster storage" paths.